File: scratch/refactor.js
const fs = require('fs');
const path = require('path');

const SRC_DIR = 'e:\\web\\src';

function processFile(filepath) {
  let content = fs.readFileSync(filepath, 'utf8');
  if (!content.includes('import { db } from "@/lib/db"')) return;

  // Add await to db.*()
  let newContent = content.replace(/(?<!await\s+)db\.([a-zA-Z0-9_]+)\(/g, 'await db.$1(');

  // Add async to export default function
  newContent = newContent.replace(/export default function/g, 'export default async function');
  // Add async to GET, POST, etc.
  newContent = newContent.replace(/export function (GET|POST|PUT|PATCH|DELETE)/g, 'export async function $1');
  
  // Clean up double async
  newContent = newContent.replace(/async async/g, 'async');

  if (newContent !== content) {
    fs.writeFileSync(filepath, newContent, 'utf8');
    console.log(`Updated ${filepath}`);
  }
}

function walk(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const filepath = path.join(dir, file);
    if (fs.statSync(filepath).isDirectory()) {
      walk(filepath);
    } else if (file.endsWith('.ts') || file.endsWith('.tsx')) {
      if (file !== 'db.ts') {
        processFile(filepath);
      }
    }
  }
}

walk(SRC_DIR);

File: scratch/refactor.py
import os
import re

SRC_DIR = r"e:\web\src"

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    if 'import { db } from "@/lib/db"' not in content:
        return

    # Find all db.method() calls and prefix with await if not already
    new_content = re.sub(r'(?<!await\s)db\.([a-zA-Z0-9_]+)\(', r'await db.\1(', content)

    # Now we need to ensure functions containing await are async
    # Simple heuristic: if it's a page or route, the default export or exported functions might need async
    # For Next.js: `export default function`, `export function GET`, etc.
    new_content = re.sub(r'export default function', r'export default async function', new_content)
    new_content = re.sub(r'export function (GET|POST|PUT|PATCH|DELETE)', r'export async function \1', new_content)
    
    # Clean up double async
    new_content = new_content.replace('async async', 'async')
    
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated {filepath}")

for root, _, files in os.walk(SRC_DIR):
    for file in files:
        if file.endswith('.ts') or file.endswith('.tsx'):
            if file != 'db.ts':
                process_file(os.path.join(root, file))
